src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -860,6 +875,8 @@ public bool CleanCacheAndDisposeIfUnused()
             List<CachedConnection> list = _idleConnections;
             List<HttpConnection> toDispose = null;
             bool tookLock = false;
+            Http2Connection http2Connection = _http2Connection;
```

Comment:
I think we need to retrieve this under the lock, otherwise we might get a stale version of it.